source/features/warning-for-disallow-edits.tsx

```diff
@@ -1,53 +1,33 @@
 import './warning-for-disallow-edits.css';
 import React from 'dom-chef';
 import {$} from 'select-dom';
-import onetime from 'onetime';
 import * as pageDetect from 'github-url-detection';
-import delegate, {DelegateEvent} from 'delegate-it';
 
 import features from '../feature-manager.js';
 import attachElement from '../helpers/attach-element.js';
 
-const getWarning = onetime(() => (
+const getWarning = (): React.JSX.Element => (
 	<div className="flash flash-error mt-3 rgh-warning-for-disallow-edits">
-		<strong>Note:</strong> Maintainers may require changes. It’s easier and faster to allow them to make direct changes before merging.
+		<strong>Note:</strong> Maintainers may require changes. It&apos;s easier and faster to allow them to make direct changes before merging.
 	</div>
-));
+);
 
-function update(checkbox: HTMLInputElement): void {
-	if (checkbox.checked) {
-		getWarning().remove();
-	} else {
-		attachElement(
-			checkbox.closest(`
-				.timeline-comment,
-				.discussion-sidebar-item > .d-inline-flex
-			`),
-			{after: getWarning},
-		);
-	}
-}
-
-function toggleHandler(event: DelegateEvent<Event, HTMLInputElement>): void {
-	update(event.delegateTarget);
-}
-
-function init(signal: AbortSignal): void | false {
+function init(): void | false {
 	const checkbox = $('input[name="collab_privs"]');
 	if (!checkbox) {
 		return false;
 	}
 
-	// TODO: Rewrite with :has selector, maybe in https://github.com/refined-github/refined-github/issues/7574
-	update(checkbox); // The sidebar checkbox may already be un-checked
-	delegate('input[name="collab_privs"]', 'change', toggleHandler, {signal});
+	attachElement(
+		checkbox.closest('.discussion-sidebar-item'),
```

Comment:
You can use `observe('.discussion-sidebar-item:has(input[name="collab_privs"])')` instead of `attachElement`, because the latter won't re-apply the fix if the sidebar is re-generated on the fly (it happens)